### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by ZohebShaikh**

### CODE REVIEW

File: src/ophyd_async/core/_detector.py [hunk 1]
```diff
@@ -65,6 +65,10 @@ class TriggerInfo(BaseModel):
     #: e.g. if num=10 and multiplier=5 then the detector will take 10 frames,
     #: but publish 2 indices, and describe() will show a shape of (5, h, w)
     multiplier: int = 1
+    #: The number of times the detector can go through a complete cycle of kickoff and
+    #: complete without needing to re-arm. This is important for detectors where the
+    #: process of arming is expensive in terms of time
+    iteration: int = 0
```
Comment (DiamondJoseph) [inline src/ophyd_async/core/_detector.py:71]: iterations or max_iterations may be slightly clearer?

**can** go through or **will**?

File: src/ophyd_async/core/_detector.py [hunk 2]
```diff
@@ -73,32 +77,36 @@ class DetectorControl(ABC):
     arming and disarming a detector
     """
 
+    _arm_status = AsyncStatus
```
Comment (DiamondJoseph) [inline src/ophyd_async/core/_detector.py:80]:
```suggestion
    _arm_status: AsyncStatus | None
```

This does require that all subclasses of DetectorControl have this field, and that it is this type. As arm is abstract, and we're leaving implementation to the implementations I'd avoid having any required fields like this.

File: src/ophyd_async/core/_detector.py [hunk 3]
```diff
@@ -73,32 +77,36 @@ class DetectorControl(ABC):
...
-        exposure: Optional[float] = None,
-    ) -> AsyncStatus:
+    def prepare(self, trigger_info: TriggerInfo) -> None:
         """
-        Arm detector, do all necessary steps to prepare detector for triggers.
-
+        Prepare the detector with the trigger Info
         Args:
-            num: Expected number of frames
+            TriggerInfo which contains
```
Comment (DiamondJoseph) [inline src/ophyd_async/core/_detector.py:91]: Update to the new prepare params

File: src/ophyd_async/core/_detector.py [hunk 4]
```diff
@@ -73,32 +77,36 @@ class DetectorControl(ABC):
...
+        """
 
-        Returns:
-            AsyncStatus: Status representing the arm operation. This function returning
-            represents the start of the arm. The returned status completing means
-            the detector is now armed.
+    @abstractmethod
+    async def arm(self):
+        """
+        Arm the detector and save the status of arm in _arm_status
```
Comment (DiamondJoseph) [inline src/ophyd_async/core/_detector.py:103]:
```suggestion
        Arm the detector
```

File: src/ophyd_async/core/_detector.py [hunk 5]
```diff
@@ -248,8 +256,13 @@ async def trigger(self) -> None:
                     trigger=DetectorTrigger.internal,
                     deadtime=None,
                     livetime=None,
+                    frame_timeout=None,
                 )
             )
+        assert self._trigger_info, "Trigger Info cannot be none after prepare"
+        assert (
+            self._trigger_info.trigger == DetectorTrigger.internal
+        ), "It can only be triggered for internal trigger"
```
Comment (DiamondJoseph) [inline src/ophyd_async/core/_detector.py:265]:
```suggestion
        ), "Cannot internally trigger with {self._trigger_info.trigger}"
```

**Review by DiamondJoseph — COMMENTED** [review]
comments on draft

**Comment by ZohebShaikh** [comment]
closed as duplicate https://github.com/bluesky/ophyd-async/pull/568

**Pull request closed without merging**